http3: flaky TestConnGoAwayFailures

=== RUN   TestConnGoAwayFailures/increased_stream_ID
    conn_test.go:234: 
        	Error Trace:	/home/runner/work/quic-go/quic-go/http3/conn_test.go:234
        	            				/home/runner/work/quic-go/quic-go/http3/conn_test.go:190
        	Error:      	Target error should be in err chain:
        	            	expected: "Application error 0x108 (remote)"
        	            	in chain: "Application error 0x100 (remote)"
        	            		"use of closed network connection"
        	Test:       	TestConnGoAwayFailures/increased_stream_ID
--- FAIL: TestConnGoAwayFailures (0.12s)

--- a/http3/conn_test.go
+++ b/http3/conn_test.go
@@ -213,6 +213,8 @@ func testConnControlStreamFailures(t *testing.T, data []byte, readErr error, exp
 		conn.handleUnidirectionalStreams(nil)
 	}()
 
+	conn.openRequestStream(context.Background(), nil, nil, true, 1000)
+
 	switch readErr {
 	case nil:
 		_, err = controlStr.Write(data)
@@ -226,8 +228,6 @@ func testConnControlStreamFailures(t *testing.T, data []byte, readErr error, exp
 		time.Sleep(scaleDuration(10 * time.Millisecond))
 		controlStr.CancelWrite(1337)
 	}
-
-	conn.openRequestStream(context.Background(), nil, nil, true, 1000)
 
 	select {
 	case <-serverConn.Context().Done():

